sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -360,25 +401,25 @@ public boolean equals(Object o) {
       if (o instanceof TimestampedValueInGlobalWindow) {
         TimestampedValueInGlobalWindow<?> that =
             (TimestampedValueInGlobalWindow<?>) o;
-        return this.timestamp.isEqual(that.timestamp) // don't compare chronology objects
-            && Objects.equals(that.pane, this.pane)
-            && Objects.equals(that.value, this.value);
+        return this.getTimestamp().isEqual(that.getTimestamp()) // don't compare chronology objects
```

Comment:
^ here's the note about the efficiency of timestamp comparisons -- at the least we should be using .isEqual(...) for all timestamp comparisons.